pkg/tsdb/influxdb/response_parser.go

```diff
@@ -45,9 +47,12 @@ func (rp *ResponseParser) Parse(buf io.ReadCloser, query *Query) plugins.DataQue
 			queryRes.Error = fmt.Errorf(result.Error)
 		}
 	}
-	queryRes.Dataframes = plugins.NewDecodedDataFrames(frames)
+	resp.Responses["A"] = queryRes
+	res := resp.Responses["A"]
+	res.Frames = frames
+	resp.Responses["A"] = res
```

Comment:
This looks a bit weird. Can you set `Frames` on `queryRes` first and then assign it to `resp`? Maybe I'm misunderstanding something?

```suggestion
	queryRes.Frames = frames
	resp.Responses["A"] = queryRes
```